Source/CarthageKit/Xcode.swift

```diff
@@ -790,10 +804,31 @@ public func buildScheme(scheme: String, withConfiguration configuration: String,
 		var argsForBuilding = argsForLoading
 		argsForBuilding.onlyActiveArchitecture = .No
 
-		var buildScheme = xcodebuildTask("build", argsForBuilding)
-		buildScheme.workingDirectoryPath = workingDirectoryURL.path!
+		let fetchDestination = { (var arguments: BuildArguments) -> SignalProducer<BuildArguments, ReactiveTaskError> in
```

Comment:
Passing `argsForBuilding` for just mutating it seems not good. How about the following?

```swift
func fetchDestination() -> SignalProducer<String?, ReactiveTaskError> {
	if sdk == .iPhoneSimulator {
		let destinationLookup = TaskDescription(launchPath: "/usr/bin/xcrun", arguments: [ "simctl", "list", "devices" ])
		return launchTask(destinationLookup)
			|> ignoreTaskData
			|> map { data in
				let string = NSString(data: data, encoding: NSStringEncoding(NSUTF8StringEncoding))!
				let regex = NSRegularExpression(pattern: "-- iOS [0-9.]+ --\\n.*?\\(([0-9A-Z]{8}-([0-9A-Z]{4}-){3}[0-9A-Z]{12})\\)", options: nil, error: nil)!

				let lastDeviceResult = regex.matchesInString(string as String, options: nil, range: NSRange(location: 0, length: (string as NSString).length)).last as? NSTextCheckingResult
				return lastDeviceResult.map { result in
					let deviceID = string.substringWithRange(result.rangeAtIndex(1))
					return "platform=iOS Simulator,id=\(deviceID)"
				}
		}
	}

	return SignalProducer(value: nil)
}

return fetchDestination()
	|> flatMap(.Concat) { destination -> SignalProducer<TaskEvent<NSData>, ReactiveTaskError> in
		if let destination = destination {
			argsForBuilding.destination = destination
			argsForBuilding.destinationTimeout = 3
		}

		var buildScheme = xcodebuildTask("build", argsForBuilding)
		buildScheme.workingDirectoryPath = workingDirectoryURL.path!

		return launchTask(buildScheme)
	}
```

And also we can use a local function here instead of a closure.